matchms/similarity/BaseSimilarityFunction.py

```diff
@@ -0,0 +1,54 @@
+from typing import List
+import numpy
+from matchms.typing import SpectrumType
+
+
+class BaseSimilarityFunction:
+    """Similarity function base class.
+    When building a custom similarity measure, inherit from this class and implement
+    the desired methods.
+    """
+    # Set key characteristics as class attributes
+    is_commutative = True
+
+    def pair(self, reference: SpectrumType, query: SpectrumType) -> float:
+        """Required: Method to calculate the similarity for one input pair.
```

Comment:
Instead of documenting required/optional methods use [from abc import abstractmethod](https://docs.python.org/3/library/abc.html#abc.abstractclassmethod) to force sub-classes to override method. Also Sphinx will decorate method with abstract keyword.

```suggestion
    @abstractmethod
    def pair(self, reference: SpectrumType, query: SpectrumType) -> float:
        """Method to calculate the similarity for one input pair.
```

Reply:
Looks nicer. Thanks.